From 8f474bfafef5d1e73bcdc048377c9251767f6595 Mon Sep 17 00:00:00 2001
From: jinghua <jinghua@marvell.com>
Date: Mon, 13 Apr 2015 14:14:47 +0300
Subject: [PATCH 0178/1240] fix: uart: move CONFIG_SYS_NS16550 to Kconfig

	CONFIG_SYS_NS16550 was compiled under the unnecessary
	dependency for CONFIG_MVEBU_UART define.

	1. Move CONFIG_SYS_NS16550 from header file to drivers/serial/Kconfig.
	2. Add CONFIG_SYS_NS16550 to A38x, ArmadaLP and A8K defconfig files.

Change-Id: I8e18b35537ac6c24ec6c14259aeef8c23efe77f4
Signed-off-by: jinghua <jinghua@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/18245
Tested-by: Star_Automation <star@marvell.com>
Reviewed-by: Omri Itach <omrii@marvell.com>
Reviewed-by: Hanna Hawa <hannah@marvell.com>
---
 configs/mvebu_armada38x_defconfig | 2 +-
 configs/mvebu_armada8k_defconfig  | 2 +-
 configs/mvebu_armadalp_defconfig  | 2 +-
 drivers/serial/Kconfig            | 8 ++++++++
 include/configs/mvebu-common.h    | 5 ++---
 5 files changed, 13 insertions(+), 6 deletions(-)

diff --git a/configs/mvebu_armada38x_defconfig b/configs/mvebu_armada38x_defconfig
index 641466a..1da0066 100644
--- a/configs/mvebu_armada38x_defconfig
+++ b/configs/mvebu_armada38x_defconfig
@@ -5,7 +5,7 @@ CONFIG_MVEBU_NAND=y
 CONFIG_MVEBU_SPI=y
 CONFIG_MVEBU_PCI=y
 CONFIG_MVEBU_MPP=y
-+S:CONFIG_MVEBU_UART=y
++S:CONFIG_SYS_NS16550=y
 +S:CONFIG_MVEBU_I2C=y
 CONFIG_CMD_BDI=y
 CONFIG_CMD_BOOTD=y
diff --git a/configs/mvebu_armada8k_defconfig b/configs/mvebu_armada8k_defconfig
index 2be4205..8acd44b 100644
--- a/configs/mvebu_armada8k_defconfig
+++ b/configs/mvebu_armada8k_defconfig
@@ -2,8 +2,8 @@ CONFIG_SPL=y
 CONFIG_SYS_EXTRA_OPTIONS="ARM64"
 +S:CONFIG_ARM=y
 +S:CONFIG_TARGET_ARMADA_8K=y
-+S:CONFIG_MVEBU_UART=y
 CONFIG_MVEBU_MPP=y
++S:CONFIG_SYS_NS16550=y
 +S:CONFIG_MVEBU_MK6_SUPPORT=y
 CONFIG_CMD_BOOTD=y
 CONFIG_CMD_RUN=y
diff --git a/configs/mvebu_armadalp_defconfig b/configs/mvebu_armadalp_defconfig
index 980f17c..edd3f10 100644
--- a/configs/mvebu_armadalp_defconfig
+++ b/configs/mvebu_armadalp_defconfig
@@ -1,7 +1,7 @@
 CONFIG_SPL=y
 +S:CONFIG_ARM=y
 +S:CONFIG_TARGET_ARMADA_LP=y
-+S:CONFIG_MVEBU_UART=y
++S:CONFIG_SYS_NS16550=y
 CONFIG_CMD_BDI=y
 CONFIG_CMD_BOOTD=y
 CONFIG_CMD_RUN=y
diff --git a/drivers/serial/Kconfig b/drivers/serial/Kconfig
index a0b6e02..3f6901b 100644
--- a/drivers/serial/Kconfig
+++ b/drivers/serial/Kconfig
@@ -5,6 +5,14 @@ config DM_SERIAL
 	  If you want to use driver model for serial drivers, say Y.
 	  To use legacy serial drivers, say N.
 
+config SYS_NS16550
+	bool "Support NS16550 for serial drivers"
+	help
+	  Choose this option to add support
+	  for NS16550 serial driver.
+	  Register base address is configured
+	  by CONFIG_SYS_NS16550_COM.
+
 config UNIPHIER_SERIAL
 	bool "UniPhier on-chip UART support"
 	depends on ARCH_UNIPHIER && DM_SERIAL
diff --git a/include/configs/mvebu-common.h b/include/configs/mvebu-common.h
index 34b1927..1b14a10 100644
--- a/include/configs/mvebu-common.h
+++ b/include/configs/mvebu-common.h
@@ -193,8 +193,7 @@
 /*********  IO Definitions **********/
 
 /* UART */
-#ifdef CONFIG_MVEBU_UART
-	#define CONFIG_SYS_NS16550
+#ifdef CONFIG_SYS_NS16550
 	#define CONFIG_SYS_NS16550_SERIAL
 	#define CONFIG_SYS_NS16550_MEM32
 	#define CONFIG_SYS_NS16550_REG_SIZE	(-4)
@@ -204,7 +203,7 @@
 	#define CONFIG_BAUDRATE			115200
 	#define CONFIG_SYS_LOADS_BAUD_CHANGE	/* allow baudrate changes */
 	/*#define CONFIG_SYS_DUART_CHAN		0*/
-#endif /* CONFIG_MVEBU_UART */
+#endif /* CONFIG_SYS_NS16550 */
 
 
 /* GPIO */
-- 
1.9.1

